screens/GridScreen.js

```diff
@@ -0,0 +1,88 @@
+import { View, Text, StyleSheet, FlatList } from 'react-native';
+import useProductList from '../Utils/hooks/useProductList';
+import GridTile from '../components/GridTile';
+import { createMaterialTopTabNavigator } from '@react-navigation/material-top-tabs';
+import { useRoute } from '@react-navigation/native';
+
+const Tab = createMaterialTopTabNavigator();
+
+function GridScreen(){
+    const route = useRoute();
+
+    function renderCategoryItem(itemData) {
```

Comment:
Please use memo, useMemo, useCallback and other methods to reduce the number of renderings of basic components. Through debugging, you can find that the number of renderings of each product item is very high. Not only solve the rendering of components in this place, please check all rendering components.